fouleggs/src/de/up/hpi/informationsystems/fouleggs/dactors/AdminSession.scala

```diff
@@ -0,0 +1,176 @@
+    log.info(s"Adding person $personId as $roleName to film $filmId")
+    val functor: ActorRef = context.system.actorOf(CastAndFilmographyFunctor.props(personId, filmId, roleName, self))
+    context.become(waitingForSuccess(functor) orElse commonBehaviour)
+  }
+
+  def waitingForSuccess(from: ActorRef): Receive = {
+    case akka.actor.Status.Success if sender == from => {
+      context.become(commonBehaviour)
+      log.info("Connected cast to film")
+
+      val empire = Dactor.dactorSelection(context.system, classOf[Film], 1)
+      val mark = Dactor.dactorSelection(context.system, classOf[Person], 1)
+
+      empire ! DefaultMessagingProtocol.SelectAllFromRelation.Request(Film.Cast.name)
+      mark ! DefaultMessagingProtocol.SelectAllFromRelation.Request(Person.Filmography.name)
+    }
+  }
+}
+
+object CastAndFilmographyFunctor {
+  def props(personId: Int, filmId: Int, roleName: String, backTo: ActorRef): Props =
+    Props(new CastAndFilmographyFunctor(personId, filmId, roleName, backTo))
+}
+
+class CastAndFilmographyFunctor(personId: Int, filmId: Int, roleName: String, backTo: ActorRef) extends Actor {
+
+  val sub1: ActorRef = context.system.actorOf(AddFilmFunctor.props(personId, filmId, roleName, self))
+  val sub2: ActorRef = context.system.actorOf(AddCastFunctor.props(personId, filmId, roleName, self))
+
+  override def receive: Receive = waitingForAck(Seq(sub1, sub2))
+
+  def waitingForAck(pending: Seq[ActorRef]): Receive = {
+    case akka.actor.Status.Success =>
+      val remainingACKs = pending.filterNot(_ == sender())
+
+      if(remainingACKs.isEmpty) {
+        backTo ! akka.actor.Status.Success
+        context.stop(self)
+      } else {
+        context.become(waitingForAck(remainingACKs))
+      }
+
+    case akka.actor.Status.Failure(e) =>
+      backTo ! akka.actor.Status.Failure(e)
+      context.stop(self)
+  }
+}
+
+object AddFilmFunctor {
+  def props(personId: Int, filmId: Int, roleName: String, backTo: ActorRef): Props =
+    Props(new AddFilmFunctor(personId: Int, filmId: Int, roleName: String, backTo: ActorRef))
+}
+
+class AddFilmFunctor(personId: Int, filmId: Int, roleName: String, backTo: ActorRef) extends Actor {
+
+  override def receive: Receive = waitingForFilmInfo orElse commonBehaviour
+
+  Dactor.dactorSelection(context.system, classOf[Film], filmId) ! DefaultMessagingProtocol.SelectAllFromRelation.Request("film_info")
+
+  def waitingForFilmInfo: Receive = {
+    case DefaultMessagingProtocol.SelectAllFromRelation.Failure(e) => fail(e)
+    case DefaultMessagingProtocol.SelectAllFromRelation.Success(relation: Relation) => {
+      context.become(waitingForInsertAck orElse commonBehaviour)
+
+      val filmInfo: Record = relation.records.get.head
+
+      val newFilmRecord: Record = Person.Filmography.newRecord(
+        Person.Filmography.filmId ~> filmId &
+        Person.Filmography.roleName ~> roleName &
+        Person.Filmography.filmName ~> filmInfo.get(Film.Info.title).get &
```

Comment:
avoid `record.get(col).get` by using the apply-method: `record(col)`: same effect, but nicer